### PR TITLE
Add keepalive and some minor fixes with disposables

### PR DESCRIPTION
**Pull request by mbonneau**

### CODE REVIEW

File: src/MessageSubject.php [hunk 1]
```diff
@@ -63,31 +68,67 @@ function (FrameInterface $frame) use ($rawDataOut) {
                             parent::onError($exception);
                         }
 
-                        // complete output stream
-                        $rawDataOut->onCompleted();
+                        $this->rawDataOut->onCompleted();
 
-                        // signal subscribers that we are done here
-                        //parent::onCompleted();
```
Comment (davidwdan) [inline src/MessageSubject.php]: Can you remove this line?

File: src/MessageSubject.php [hunk 2]
```diff
@@ -63,31 +68,67 @@ function (FrameInterface $frame) use ($rawDataOut) {
...
+
+                        $this->rawDataDisp->dispose();
                         return;
                 }
             },
             !$this->mask
         );
 
-        $this->rawDataDisp = $this->rawDataIn->subscribe(
-            function ($data) use ($messageBuffer) {
```
Comment (davidwdan) [inline src/MessageSubject.php:78]: Consider using `[$messageBuffer, 'onData']`

File: src/MessageSubject.php [hunk 3]
```diff
@@ -63,31 +68,67 @@ function (FrameInterface $frame) use ($rawDataOut) {
...
-                $messageBuffer->onData($data);
-            },
-            function (\Exception $exception) {
```
Comment (davidwdan) [inline src/MessageSubject.php:81]: `[parent, 'onError']` and the other places below .

**Comment by mbonneau** [comment]
ping @davidwdan

**Pull request merged**